fe/fe-core/src/main/java/com/starrocks/sql/analyzer/UpdateAnalyzer.java

```diff
@@ -87,27 +87,29 @@ public static void analyze(UpdateStmt updateStmt, ConnectContext session) {
             }
         }
 
-        if (session.getSessionVariable().getPartialUpdateMode().equals("column")) {
-            // use partial update by column
-            updateStmt.setUsePartialUpdate();
-        } else if (session.getSessionVariable().getPartialUpdateMode().equals("auto")) {
-            // decide by default rules
-            if (updateStmt.getWherePredicate() == null) {
-                if (checkIfUsePartialUpdate(assignmentList.size(), table.getBaseSchema().size())) {
-                    // use partial update if:
-                    // 1. Columns updated are less than 4
-                    // 2. The proportion of columns updated is less than 30%
-                    // 3. No where predicate in update stmt
-                    updateStmt.setUsePartialUpdate();
-                } else {
+        if (table.isOlapTable()) {
+            if (session.getSessionVariable().getPartialUpdateMode().equals("column")) {
+                // use partial update by column
+                updateStmt.setUsePartialUpdate();
+            } else if (session.getSessionVariable().getPartialUpdateMode().equals("auto")) {
+                // decide by default rules
+                if (updateStmt.getWherePredicate() == null) {
+                    if (checkIfUsePartialUpdate(assignmentList.size(), table.getBaseSchema().size())) {
+                        // use partial update if:
+                        // 1. Columns updated are less than 4
+                        // 2. The proportion of columns updated is less than 30%
+                        // 3. No where predicate in update stmt
+                        updateStmt.setUsePartialUpdate();
+                    } else {
+                        throw new SemanticException("must specify where clause to prevent full table update");
+                    }
+                }
+            } else {
+                // when var `partial_update_mode` == row, use full columns update instead of partial update
+                if (updateStmt.getWherePredicate() == null) {
                     throw new SemanticException("must specify where clause to prevent full table update");
                 }
             }
-        } else {
```

Comment:
else {
   if (updateStmt.getWherePredicate() == null) {
         throw new SemanticException("must specify where clause to prevent full table update");
     }
}